### PR TITLE
Open addressed hashing for static aggregations.

### PR DESCRIPTION
**Pull request by cpwright**

### CODE REVIEW

**Review by cpwright — COMMENTED** [review]
Viewed placeholder.

**Review by cpwright — COMMENTED** [review]
Charles place holder.

**Review by cpwright — COMMENTED** [review]
Placeholder.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/by/OperatorAggregationStateManagerOpenAddressedBase.java [hunk 1]
```diff
@@ -29,28 +28,28 @@
...
-    protected final FlatArraySource[] mainKeySources;
+    protected final ColumnSource[] mainKeySources;
 
     protected OperatorAggregationStateManagerOpenAddressedBase(ColumnSource<?>[] tableKeySources, int tableSize,
             double maximumLoadFactor) {
         this.tableSize = tableSize;
         Require.leq(tableSize, "tableSize", MAX_TABLE_SIZE);
         Require.gtZero(tableSize, "tableSize");
         Require.eq(Integer.bitCount(tableSize), "Integer.bitCount(tableSize)", 1);
+        Require.gtZero(maximumLoadFactor, "maximumLoadFactor");
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/by/OperatorAggregationStateManagerOpenAddressedBase.java]: Use `inRange`.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/staticopenagg/gen/StaticAggOpenHasherByteChar.java [hunk 2]
```diff
@@ -79,63 +79,41 @@ private static int hash(byte k0, char k1) {
...
-                                moveMainDest.set(endMovePointer--, tableLocation);
-                            }
-                        }
-                        break;
-                    } else {
-                        Assert.neq(tableLocation, "tableLocation", lastTableLocation, "lastTableLocation");
-                        tableLocation = (tableLocation + 1) & (tableSize - 1);
+    protected void rehashInternal() {
+        final int oldSize = tableSize >> 1;
+        final byte[] destArray0 = new byte[tableSize];
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/staticopenagg/gen/StaticAggOpenHasherByteChar.java]: destKey?

**Review by rcaudy — APPROVED** [review]

**Pull request merged**